### ISSUE TITLE
M.2 Drive detected by script but not showing in DSM or able to create volume

### ISSUE THREAD

**Issue by alirz1** [007revad/Synology_enable_M2_volume#185]
Im running this on a HP elitedesk 800 G3 mini PC. It has  a SSD and a M.2 nvme drive.
I installed xpenology(DS918+) on it. Only the sata SSD drive is currently showing in DSM.

I've tried running the 3 scripts, the HDD DB, Synology_enable_M2_volume, Synology_M2_volume.

The scripts show me the M.2 drive being detected. But fails to show up in DSM storage manager or even create volume using the script.
Here is the message i get:

`Running from: /volume1/data/Synology_M2_volume-2.0.27/syno_create_m2_volume.sh

nvme0n1 M.2 Drive nvme0n1 - SK Hynix PC300 NVMe SK hynix 256GB

Unused M.2 drives found: 1

1) SHR 1
2) Basic
3) JBOD
Select the RAID type: 2
You selected basic

You have a choice of Multi Volume or Single Volume Storage Pool
 - Multi Volume Storage Pools allow creating multiple volumes and
   allow you to over provision to make the NVMe drive(s) last longer.
 - Single Volume Storage Pools are easier to recover data from
   and perform slightly faster.

1) Multi Volume (DSM 7 default)
2) Single Volume
Select the storage pool type: 1
You selected Multi Volume storage pool

1) nvme0n1
Select the M.2 drive: 1
You selected nvme0n1

Do you want perform a drive check? [y/n]
n

Ready to create Basic storage pool using nvme0n1
Type yes to continue. Type anything else to quit.
yes

Starting creation of the storage pool.
0 synostgpool failed to create storage pool!
`

**Comment by 007revad** [comment]
I just installed 1 NVMe drive in a DS720+ and I'm getting the same `0 synostgpool failed to create storage pool!` error. When I went to storage manager it showed the storage could be online assembled, but when clicking on "online assemble" it fails.

I'm trying to work out why it works with 2 or more NVMe drives but fails with 1. I have noticed that the single NVMe drive ends up with an extra partition.
